src/utils/save_game/save_game.cpp

```diff
@@ -0,0 +1,66 @@
+#include "save_game.h"
+#include "../../creature/creature.h"
+#include "../../creature/main_character/main_character.h"
+#include <iostream>
+#include <fstream>
+
+void SaveGame(MainCharacter& main_characher, Creature& creature, const std::string& filename)
+{
+	std::ofstream outputFile(filename);
+
+	if (outputFile.is_open())
+	{
+		outputFile << "Ability Scores:" << ::std::endl;
+		outputFile << "==============================" << std::endl;
+		outputFile << "HP: " << creature.get_hp() << std::endl;
+		outputFile << "AC: " << creature.get_ac() << std::endl;
+		outputFile << "Speed: " << creature.get_speed() << std::endl;
+		outputFile << "Strength: " << creature.get_ability_score(0) << std::endl;
+		outputFile << "Dexterity: " << creature.get_ability_score(1) << std::endl;
+		outputFile << "Constitution: " << creature.get_ability_score(2) << std::endl;
+		outputFile << "Intelligence: " << creature.get_ability_score(3) << std::endl;
+		outputFile << "Wisdom: " << creature.get_ability_score(4) << std::endl;
+		outputFile << "Charisma: " << creature.get_ability_score(5) << std::endl;
+		outputFile << "==============================" << std::endl;
+		outputFile << "Ability Scores for: " << main_characher.get_name() << std::endl;
+		outputFile << "==============================" << std::endl;
+		outputFile << "HP: " << main_characher.get_hp() << std::endl;
+		outputFile << "AC: " << main_characher.get_ac() << std::endl;
+		outputFile << "Speed: " << main_characher.get_speed() << std::endl;
+		outputFile << "Strength: " << main_characher.get_ability_score(0) << std::endl;
+		outputFile << "Dexterity: " << main_characher.get_ability_score(1) << std::endl;
+		outputFile << "Constitution: " << main_characher.get_ability_score(2) << std::endl;
+		outputFile << "Intelligence: " << main_characher.get_ability_score(3) << std::endl;
+		outputFile << "Wisdom: " << main_characher.get_ability_score(4) << std::endl;
+		outputFile << "Charisma: " << main_characher.get_ability_score(5) << std::endl;
+		outputFile << "==============================" << std::endl;
+
+		outputFile.close();
+
+		std::cout << "Game saved successfully to " << filename << std::endl;
+	}
+	else
+	{
+		std::cerr << "Unable to open file: " << filename << std::endl;
+	}
+}
+void SaveSettings(int soundVolume, int textSize, int textSpeed, int weight_shoice, const std::string& filename)
```

Comment:
I can see this function does the same as the above, only it saves to a different file. I would advocate that these be one function that saves to one file, like the issue describes.